shells/planner-shell/single-user-context.js

```diff
@@ -192,15 +192,14 @@ export const SingleUserContext = class {
     // TODO(sjmiles): cache and return promises in case of re-entrancy
     let promise = this.pendingStores[id];
     if (!promise) {
-      promise = new Promise(async (resolve) => {
-        const store = await context.findStoreById(id);
-        if (store) {
-          resolve(store);
-        } else {
-          const store = await context.createStore(type, name, id, tags);
-          resolve(store);
+      // eslint-disable-next-line no-async-promise-executor
+      promise = async () => {
```

Comment:
This looks like changing behaviour. This used to assign a promise, but now you assign an async function. What I think you want instead is to define and call this function straight away, assigning the result of the call to the promise variable:

```
promise = (async () => {
  let store = context.findStoreById(id);
  if (!store) {
    store = await context.createStore(type, name, id, tags);
  }
  return store;
})();
```

Reply:
indeed, thanks for the fix.